tests/Properties/Primitives.hs

```diff
@@ -0,0 +1,45 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+module Properties.Primitives (tests) where
+
+import Test.Framework (Test, testGroup)
+import Test.Framework.Providers.QuickCheck2 (testProperty)
+import Test.QuickCheck
+
+import Control.Applicative ((<$>), (<*>))
+
+import Data.BTree.Primitives
+
+import Data.Int
+
+import Properties.Primitives.Height () -- Arbitrary instance of Height
+import Properties.Primitives.Index ()  -- Arbitrary instance of Index
+import Properties.Primitives.Ids ()    -- Arbitrary instance of NodeId
+
+import Properties.Utils (testBinary)
+
+tests :: Test
+tests = testGroup "Primitives.Ids"
+    [ testProperty "binary leafNode" prop_binary_leafNode
+    , testProperty "binary indexNode" prop_binary_indexNode
+    , testProperty "binary tree" prop_binary_tree
+    ]
+
+instance (Key k, Arbitrary k, Arbitrary v) => Arbitrary (Node 'Z k v) where
+    arbitrary = Leaf <$> arbitrary
+
+instance (Key k, Arbitrary k, Arbitrary v) => Arbitrary (Node ('S height) k v) where
```

Comment:
The `Arbitrary` instance of `Index` doesn't respect any limits on the number of keys/subtrees. Having arbitrary and too large indices for the tests here is fine, but I think it's getting a bit awkward that it's propagating to other instances. It's probably better to use explicit generators instead of `Arbitrary` instances. Let's leave it for now. I'll create an issue for this to deal with it later.